src/module/hooks/CreateChatMessage.ts

```diff
@@ -156,6 +159,60 @@ class CreateChatMessage {
         }
         return Settings.setCounter(counter);
     }
+
+    /**
+     * Parses embedded rolls to make them JSONs
+     * 
+     * @param message - message send in chat
+     * @param user - author of the message
+     */
+    private async _parseEmbeddedRolls (message: any, user: any): Promise<Array<number>> {
+        const regexRoll = /roll=\"(.*?)\"/g;
+        const matches = [...message.matchAll(regexRoll)];
+        const dieType = SadnessChan.getDieType();
+        if (!(matches && matches.length > 0)) return [];
+
+        let allRecentRolls = this._getZeroArray(dieType + 1);
+        matches.forEach((element:any) => {
+            try {
+                const parsedEmbedded = JSON.parse(decodeURIComponent(element[1]));
+                const recentRolls = this._extractEmbeddedRolls(parsedEmbedded, user);
+                recentRolls.forEach((element: any, index: number) => {
+                    allRecentRolls[index] = element;
+                })
+            } catch (error) {
+                return []
+            }
+        });
+        
+        await this._updateDiceRolls(allRecentRolls, this._prepareUserDataForStorage(user));
+        return allRecentRolls;
+    }
+
+    /**
+     * Extracts rolls from the embedded JSON structure
+     * 
+     * @param messageJSON - parsed message
+     * @param user - owner of the message
+     */
+    private _extractEmbeddedRolls (messageJSON: any, user: any): Array<number> {
+        const terms = messageJSON.terms;
+        const dieType = SadnessChan.getDieType();
+        const recentRolls = this._getZeroArray(dieType + 1);
+        if (!terms) return;
+        
+        terms.forEach((term: any) => {
+            if (term == '+' || term.faces != dieType) return;
+
+            term.results.forEach((element: any) => {
+                recentRolls[element.result] += 1;
+            });
```

Comment:
Could be a one-liner:
`term.results.foreach(element: any => recentRolls[element.result] += 1)`